spec/rubocop/cop/packaging/require_hardcoding_lib_spec.rb

```diff
@@ -5,45 +5,9 @@
 
   let(:project_root) { RuboCop::ConfigLoader.project_root }
 
-  context "when `require` call lies outside spec/" do
-    let(:filename) { "#{project_root}/spec/foo_spec.rb" }
-    let(:source) { "require '../lib/foo.rb'" }
-
-    it "registers an offense" do
-      expect_offense(<<~RUBY, filename)
-        #{source}
-        #{"^" * source.length} #{message}
-      RUBY
-
-      expect_correction(<<~RUBY)
-        require "foo.rb"
-      RUBY
-    end
-  end
-
-  context "when `require` call after using `unshift` lies outside spec/" do
-    let(:filename) { "#{project_root}/tests/foo/bar.rb" }
-    let(:source) { <<~RUBY.chomp }
-      $:.unshift('../../lib')
-      require '../../lib/foo/bar'
-    RUBY
-
-    it "registers an offense" do
-      expect_offense(<<~RUBY, filename)
-        #{source}
-        #{"^" * 27} #{message}
-      RUBY
-
-      expect_correction(<<~RUBY)
-        $:.unshift('../../lib')
-        require "foo/bar"
-      RUBY
-    end
-  end
-
-  context "when `require` call uses File#expand_path method with __FILE__" do
```

Comment:
I expect these tests to be still working, really. Might need some fixing but in the end, it should work, too (even as-is, from a quick POV). Because we're essentially just generalizing the entire set of calls and the above tests are really just a subset of those. So I expect them to work + I expect new tests to be added (the generalized one, mentioned in the issue, for example).